src/main/java/bio/terra/service/dataset/AssetDao.java

```diff
@@ -34,6 +38,14 @@ public AssetDao(DataRepoJdbcConfiguration jdbcConfiguration) {
                 .collect(Collectors.toList());
     }
 
+    /**
+     * Create a new AssetSpecification. If a DuplicateKeyException is thrown,
+     * catch it and throw an InvalidAssetException
```

Comment:
Nit: Maybe make this a little more reader-friendly? If you try to create an asset with the same name as an existing one for the same dataset, this method throws an InvalidAssetException. Or something like that. I think that might be a little better than talking about a DuplicateKeyException specifically.